### PR TITLE
fix: patch zero balance when getBalance query times out

### LINKED ISSUE #6590: [Bug] Balance amount displays `0` for several seconds and back the correct balance in a repeated sequence
**Describe the bug**
On the Home screen, I can see how balance is set to `0` for some seconds and then got back to my current balance. This happens very often (minimum once every 1 or 2 minutes).
I discard an API issue as I can see how it's working fine in production, with the same API and RPC node.

**Screenshots**

https://github.com/MetaMask/metamask-mobile/assets/54408225/f95fc28a-c480-4e07-8911-fbd238e5b0ac

**To Reproduce**
_Steps to reproduce the behavior_
1. Go to Home screen
2. Select Mainnet
3. Wait
4. See balance is set to `0` 
5. See balance is changed back to your current balance
6. See balance is set to `0`
(this sequence keeps repeating)

**Expected behavior**
Balance should not be `0` if I have funds

**Smartphone (please complete the following information):**
 - Device: Pixel 6
 - OS: Android
 - App Version: happening on current `main` (last commit https://github.com/MetaMask/metamask-mobile/commit/1b79574407323fc7bac30d1ac06190d8f0017644)

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

### PR DESCRIPTION
**Pull request by NicolasMassart**
## Development & PR Process
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

## Description

Main token (eth) balance shows zero balance when the getBalance query times out as the store is updated to the 0x0 default value instead of not being updated at all. 

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_

This randomly shows a zero balance making the wallet display frightening zero ETH and show the "Add crypto to get started" button. Button only disapear and balance is ok when a new successful call is done. So you have the wrong state for 10s between two polls.

_2. What is the improvement/solution?_

Return an error instead of 0x0 as a balance when an error actually happen. This requires an update in `patches/@metamask+assets-controllers+5.0.0.patch`

## Testing

You can trigger timeouts by reducing timeout value in `node_modules/@metamask/controller-utils/dist/util.js` at Line 194:

`function safelyExecuteWithTimeout(operation, logError = false, timeout = 500) {`
--> `function safelyExecuteWithTimeout(operation, logError = false, timeout = 1) {`

## Issue

fixes MetaMask/metamask-mobile#6590
fixes MetaMask/mobile-planning#1079
fixes MetaMask/mobile-planning#1071

## Checklist

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

File: patches/@metamask+assets-controllers+5.0.0.patch [hunk 1]
```diff
@@ -65,8 +67,8 @@ index 130e3dc..912c84c 100644
 +     */
 +    getBalanceFromChain(address) {
 +        return __awaiter(this, void 0, void 0, function* () {
-+            let balance = '0x0';
-+            yield (0, controller_utils_1.safelyExecuteWithTimeout)(() => __awaiter(this, void 0, void 0, function* () {
++            let balance;
++            const result = yield (0, controller_utils_1.safelyExecuteWithTimeout)(() => __awaiter(this, void 0, void 0, function* () {
```
Comment (tommasini) [inline patches/@metamask+assets-controllers+5.0.0.patch:71]: If we are not changing the util function `safelyExecuteWithTimeout` and only initiating the balance without a default '0x0' value, the result variable that has the return of the `safelyExecuteWithTimeout`  is not being used.

**Review by tommasini — CHANGES_REQUESTED** [review]
Left a comment! 
I just see a possible edge case, if the get balance fails and we don't have a default value, I can see the app crashing creating a new wallet for example, unless it's handled on other place that I'm not seeing

File: patches/@metamask+assets-controllers+5.0.0.patch (hunk 1 above)
Comment (NicolasMassart) [reply patches/@metamask+assets-controllers+5.0.0.patch:71]: Sorry I forgot to remove it...

**Comment by NicolasMassart** [comment]
> I just see a possible edge case, if the get balance fails and we don't have a default value, I can see the app crashing creating a new wallet for example, unless it's handled on other place that I'm not seeing

[Each new account balance state is initialised with 0x0](https://github.com/MetaMask/core/blob/dc9eb67eed23375bdcaa9db2ddfadd995d81f397/packages/assets-controllers/src/AccountTrackerController.ts#L69-L71), so it can not be undefined in state unless the state is broken, but then we have other issues...

**Review by tommasini — DISMISSED** [review]
Amazing! LGTM!

**Comment by NicolasMassart** [comment]
Fix time tracking: 3d

**Review by tommasini — APPROVED** [review]
LGTM!!!!

**Review by tommasini — DISMISSED** [review]
LGTM!!!!

**Comment by chrisleewilcox** [comment]
@NicolasMassart I changed the base branch back to release/7.2.0 but there were merge conflicts with a few files.  I then changed the base branch back to main for now.  At least for this release we'll continue working out of the release branch.  Can you take a look at the merge conflicts against release branch?  I'll retest once the conflicts are resolved.

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request merged**